### DISCUSSION TITLE
Type A port on a Pico

### DISCUSSION THREAD

**Discussion by ebadger** [hathach/tinyusb#1840] [Q&A]
Curious to know if it's possible to assign other GPIO to D+/D-?
I'd like to connect a type A port to GPIO 0,1.  
I'd like to run in host mode and accept input from a USB keyboard with a type A connector while still powering the device through the micro USB.

Possible?

**Comment by byteit101** [comment]
Not directly, but you can using the PIO: https://www.cnx-software.com/2022/02/16/raspberry-pi-rp2040-gains-an-extra-usb-port-through-pio/
